napari/utils/translations.py

```diff
@@ -50,11 +50,19 @@ def _get_display_name(
             else _DEFAULT_LOCALE
         )
         loc = babel.Locale.parse(locale)
-        dislay_name = loc.get_display_name(display_locale).capitalize()
+        display_name = loc.get_display_name(display_locale)
+        if display_name is not None:
+            display_name = display_name.capitalize()
+        else:
+            # to allow raising error inside try: block
+            raise RuntimeError(
+                f"Could not find {display_locale}"
+            )  # ruff: noqa: TRY301
+
```

Comment:
maybe everything after import should be moved to else clause. 

Or just use multiple return pattern?

This code is not readable to me.

Reply:
Thanks - agree this wasn't the best! How about now? I've used a `try: ... except: ... else: ...` pattern to make the code mroe readable, and remove the need for the ruff noqa.